### PR TITLE
Alerting: Fix incorrect warning for OnCall integration

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fix a regression  showing a warning on the oncall receiver metadata.
The problem comes from the wrong assumption that [we have this settings](https://github.com/grafana/grafana/blob/c4e0baa95f67aeda4972771b83b3e543db309d98/public/app/features/alerting/unified/components/receivers/grafanaAppReceivers/useReceiversMetadata.ts#L47C54-L47C74) field in the receivers response, and we haven't. The api used in here `/api/v1/notifications/receivers` does not return this field.

As we are moving to the k8s api for Grafana managed alerts, I think best option is to return the basic metadata until we have this settings field in the new endpoint.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes #
This issue is in this [escalation](https://github.com/grafana/support-escalations/issues/12208)

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/contact-points/__snapshots__/useContactPoints.test.tsx.snap [hunk 1]
```diff
@@ -39,7 +39,6 @@ exports[`useContactPoints should return contact points with status 1`] = `
           },
         },
       ],
-      "provisioned": false,
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/contact-points/__snapshots__/useContactPoints.test.tsx.snap:42]: now, in this test, the data is from the alert manager config , where we don't have this field

File: public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx [hunk 2]
```diff
@@ -296,13 +296,14 @@ export function useContactPointsWithStatus({
...
       ...result,
-      contactPoints: result.data
-        ? enhanceContactPointsWithMetadata({
-            notifiers: cloudNotifierTypes,
-            contactPoints: result.data.alertmanager_config.receivers ?? [],
-            alertmanagerConfiguration: result.data,
-          })
-        : [],
+      contactPoints:
+        result.data && !isGrafanaAlertmanager // I needed to add this condition to avoid the error in tests, it seems that the skip is not working as expected
```
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/components/contact-points/useContactPoints.tsx:300]: without this doble check (`!isGrafanaAlertManager` in here and in the skip option), tests were not skipping, and the result used was from the alert manager configuration....

File: public/app/features/alerting/unified/components/receivers/grafanaAppReceivers/useReceiversMetadata.ts [hunk 3]
```diff
@@ -30,6 +30,10 @@ export function getOnCallMetadata(
     return onCallReceiverMeta;
   }
 
+  if (!receiver.settings?.url) {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/receivers/grafanaAppReceivers/useReceiversMetadata.ts:33]: Can we add a TODO here and a comment explaining why we do this? :)

**Comment by gillesdemey** [comment]
The code itself looks good; I'd still love to test it in an ephemeral instance so I'm trying to deploy it :D

**Comment by gillesdemey** [comment]
After removing the integration on the OnCall side but keeping it as a contact point it seems to show up blank 🤔 

image

**Comment by soniaAguilarPeiron** [comment]
> After removing the integration on the OnCall side but keeping it as a contact point it seems to show up blank 🤔
💯

**Comment by soniaAguilarPeiron** [comment]
> After removing the integration on the OnCall side but keeping it as a contact point it seems to show up blank 🤔
> 
> I think we should figure out how we can present this in a better way if it's related to the recent changes to contact points.
> 
> image


The problem is that if we remove the integration on the oncall side, the BE is returning this receiver with only the name field...nothing else:

Screenshot 2024-09-13 at 09 24 28

**Comment by tomratcliffe** [comment]
I think that's _probably_ to be expected? We need an empty state for contact points when they don't have any integrations anyway - we could handle that soon in another PR.

**Comment by soniaAguilarPeiron** [comment]
> state for contact points when they don't have any integrations anyway - we could handle that soon in another PR.

this is not the state endpoint but the contact points endpoint

**Review by gillesdemey — APPROVED** [review]
LGTM 🚀

**Pull request merged**